src/containers/WorkspaceView.tsx

```diff
@@ -1,50 +1,184 @@
+    switch (page) {
+        case 'recordings': return {
+            workspaceName,
+            page
+        }
+        case 'recording': return {
+            workspaceName,
+            page,
+            recordingId: pathList[3]
+        }
+        case 'sorting': return {
+            workspaceName,
+            page,
+            recordingId: pathList[3] || '',
+            sortingId: pathList[4] || ''
+        }
+        default: return {
+            workspaceName,
+            page: 'recordings'
+        }
+    }
+}
+
+var queryString = (params: {[key: string]: string}) => {
+    const keys = Object.keys(params)
+    if (keys.length === 0) return ''
+    return '?' + (
+        keys.map((key) => {
+            return encodeURIComponent(key) + '=' + encodeURIComponent(params[key])
+        }).join('&')
     )
 }
 
+const locationFromRoute = (route: WorkspaceRoute, workspaceInfo: WorkspaceInfo) => {
+    const queryParams: {[key: string]: string} = {}
+    if (workspaceInfo.feedUri) {
+        queryParams['feed'] = workspaceInfo.feedUri
+    }
+    switch(route.page) {
+        case 'recordings': return {
+            pathname: `/${route.workspaceName}`,
+            search: queryString(queryParams)
+        }
+        case 'recording': return {
+            pathname: `/${route.workspaceName}/recording/${route.recordingId}`,
+            search: queryString(queryParams)
+        }
+        case 'sorting': return {
+            pathname: `/${route.workspaceName}/sorting/${route.recordingId}/${route.sortingId}`,
+            search: queryString(queryParams)
+        }
+    }
+}
+
+const WorkspaceView: FunctionComponent<Props> = ({ width, height, sortings, recordings, onDeleteRecordings, workspaceInfo, plugins }) => {
+    const history = useHistory()
+    const location = useLocation()
+    const route = routeFromLocation(location)
+    if (route.workspaceName !== workspaceInfo.workspaceName) {
+        throw Error(`Mismatch in workspace name: ${route.workspaceName} <> ${workspaceInfo.workspaceName}`)
+    }
+    const workspaceRouteDispatch = useMemo(() => ((a: WorkspaceRouteAction) => {
+        const route = routeFromLocation(history.location)
+        let newRoute: WorkspaceRoute | null = null
+        switch(a.type) {
+            case 'gotoRecordingsPage': newRoute = {
+                page: 'recordings',
+                workspaceName: route.workspaceName
+            }; break;
+            case 'gotoRecordingPage': newRoute = {
+                page: 'recording',
+                recordingId: a.recordingId,
+                workspaceName: route.workspaceName
+            }; break;
+            case 'gotoSortingPage': newRoute = {
+                page: 'sorting',
+                recordingId: a.recordingId,
+                sortingId: a.sortingId,
+                workspaceName: route.workspaceName
+            }; break
+        }
+        if (newRoute) {
+            history.push(locationFromRoute(newRoute, workspaceInfo))
```

Comment:
push the new route to the history, which will trigger navigating to the new location within the app.